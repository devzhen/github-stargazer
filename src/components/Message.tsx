import React from 'react';
import styled from 'styled-components/native';

interface MessageProps {
  message: string;
  onPress?: () => void;
  offsetTop?: number;
  offsetBottom?: number;
  size?: number;
  alignment?: string;
  weight?: string | number;
  color?: string;
  bgColor?: string;
  clientTop?: number;
  clientLeft?: number;
  clientRight?: number;
  clientBottom?: number;
  lineHeight?: number;
}

const Message = ({
  message,
  offsetTop,
  offsetBottom,
  onPress,
  size,
  alignment,
  color,
  weight,
  bgColor,
  clientTop,
  clientLeft,
  clientRight,
  clientBottom,
  lineHeight,
}: MessageProps) => (
  <Container
    offsetTop={offsetTop}
    offsetBottom={offsetBottom}
    bgColor={bgColor}
    clientTop={clientTop}
    clientLeft={clientLeft}
    clientRight={clientRight}
    clientBottom={clientBottom}
  >
    <Text
      size={size}
      alignment={alignment}
      weight={weight}
      color={color}
      onPress={onPress}
      lineHeight={lineHeight}
    >
      {message}
    </Text>
  </Container>
);

Message.defaultProps = {
  offsetTop: 0,
  offsetBottom: 0,
  size: 14,
  onPress: () => {},
  alignment: 'center',
  weight: 'bold',
  color: '#eb9e38',
  message: '',
};

const Container = styled.View<{
  offsetTop?: number;
  offsetBottom?: number;
  bgColor?: string;
  clientTop?: number;
  clientLeft?: number;
  clientRight?: number;
  clientBottom?: number;
}>`
  margin-top: ${(props) => props.offsetTop};
  margin-bottom: ${(props) => props.offsetBottom};
  align-items: center;
  justify-content: center;
  ${(props) => props.bgColor && `background-color: ${props.bgColor}`};
  padding-top: ${(props) => props.clientTop || 0};
  padding-left: ${(props) => props.clientLeft || 0};
  padding-right: ${(props) => props.clientRight || 0};
  padding-bottom: ${(props) => props.clientBottom || 0};
`;

const Text = styled.Text<{
  size?: number;
  color?: string;
  alignment?: string;
  weight?: string | number;
  lineHeight?: number;
}>`
  font-size: ${(props) => props.size};
  color: ${(props) => props.color};
  text-align: ${(props) => props.alignment};
  font-weight: ${(props) => props.weight};
  ${(props) => props.lineHeight && `line-height: ${props.lineHeight}`};
`;

export default Message;
