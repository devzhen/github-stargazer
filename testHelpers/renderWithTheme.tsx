import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import 'jest-styled-components';

import theme from '@theme/index';

const renderWithTheme = (component: any) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export default renderWithTheme;
