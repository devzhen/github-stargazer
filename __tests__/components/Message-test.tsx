import React from 'react';
import 'jest-styled-components';

import Message from '@components/Message';

// @ts-ignore
import renderWithTheme from '@testHelpers/renderWithTheme';

describe('Message common component', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Message message="Please, provide you credentials." />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with offsetTop', () => {
    const tree = renderWithTheme(
      <Message message="Please, provide you credentials." offsetTop={22} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with offsetBottom', () => {
    const tree = renderWithTheme(
      <Message message="Please, provide you credentials." offsetBottom={22} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
