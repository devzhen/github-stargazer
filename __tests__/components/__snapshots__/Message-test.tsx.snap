// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message common component renders correctly 1`] = `
<View
  offsetBottom={0}
  offsetTop={0}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 0,
        "marginTop": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      },
    ]
  }
>
  <Text
    alignment="center"
    color="#eb9e38"
    onPress={[Function]}
    size={14}
    style={
      Array [
        Object {
          "color": "#eb9e38",
          "fontSize": 14,
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
    weight="bold"
  >
    Please, provide you credentials.
  </Text>
</View>
`;

exports[`Message common component renders correctly with offsetBottom 1`] = `
<View
  offsetBottom={22}
  offsetTop={0}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 22,
        "marginTop": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      },
    ]
  }
>
  <Text
    alignment="center"
    color="#eb9e38"
    onPress={[Function]}
    size={14}
    style={
      Array [
        Object {
          "color": "#eb9e38",
          "fontSize": 14,
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
    weight="bold"
  >
    Please, provide you credentials.
  </Text>
</View>
`;

exports[`Message common component renders correctly with offsetTop 1`] = `
<View
  offsetBottom={0}
  offsetTop={22}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 0,
        "marginTop": 22,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      },
    ]
  }
>
  <Text
    alignment="center"
    color="#eb9e38"
    onPress={[Function]}
    size={14}
    style={
      Array [
        Object {
          "color": "#eb9e38",
          "fontSize": 14,
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
    weight="bold"
  >
    Please, provide you credentials.
  </Text>
</View>
`;
